src/uct/ib/base/ib_device.c

```diff
@@ -584,7 +582,12 @@ uct_ib_device_init_nb_close_ctx(int fd, uct_ib_device_nb_close_ctx **ctx_p)
     ucs_status_t status;
     int ret;
 
-    getrlimit(RLIMIT_NOFILE, &nofile);
+    ret = getrlimit(RLIMIT_NOFILE, &nofile);
+    if (ret) {
+        ucs_error("getrlimit failure");
+        status = UCS_ERR_IO_ERROR;
+        goto err;
+    }
```

Comment:
1. let's move before line 599
2. if fail to read limit, let's print warn and use 1024 by default